File: pom-experiments/pomtest3/application/src/main/java/com/stulsoft/poc/pomtest3/app/Application.java
/*
 * Copyright (c) 2018. Yuriy Stul
 */

package com.stulsoft.poc.pomtest3.app;

import com.stulsoft.poc.pomtest3.common.Utils;

/**
 * @author Yuriy Stul
 */
public class Application {
    public static void main(String[] args) {
        System.out.println("==>main");
        System.out.printf("foo=%s%n", Utils.foo());
        System.out.println("<==main");
    }
}

File: json-path-2-attr/src/main/scala/com/stulsoft/json/Application.scala
/*
 * Copyright (c) 2020. Yuriy Stul
 */

package com.stulsoft.json

import com.typesafe.scalalogging.StrictLogging

import scala.collection.mutable.ArrayBuffer

/**
 * @author Yuriy Stul
 */
object Application extends App with StrictLogging {
  logger.info("==>Application")

  test2()
  test3()

  def test2(): Unit = {
    logger.info("==>test2")
    val paths = ArrayBuffer.empty[(String, Any)]
    PathBuilder.buildPath(Utils.jsonMapFromResourse("test2.json"), "", paths)
    logger.debug("paths: {}", paths)
    logger.debug(" ")
    logger.debug("for d: {}",PathBuilder.findPathToAttr(paths, "d"))
    logger.debug(" ")
    logger.debug("for a: {}",PathBuilder.findPathToAttr(paths, "a"))
  }

  def test3(): Unit = {
    logger.info("==>test3")
    val paths = ArrayBuffer.empty[(String, Any)]
    PathBuilder.buildPath(Utils.jsonMapFromResourse("test3.json"), "", paths)
    logger.debug("paths: {}", paths)
    logger.debug(" ")
    logger.debug("for d: {}",PathBuilder.findPathToAttr(paths, "d"))
    logger.debug(" ")
    logger.debug("for a: {}",PathBuilder.findPathToAttr(paths, "a"))
  }

}
